ipatests/test_xmlrpc/test_automember_plugin.py

```diff
@@ -220,13 +212,112 @@ def test_create_with_nonexistent_hostgroup(self, automember_hostgroup,
                 reason=u'hostgroup "%s" not found' % hostgroup1.cn)):
             command()
 
-    def test_delete_with_nonexistent_hostgroup(self, automember_hostgroup,
-                                               hostgroup1):
-        """ Try to delete a rule with non-existent group """
-        hostgroup1.ensure_missing()
-        command = automember_hostgroup.make_delete_command()
-        with raises_exact(errors.NotFound(
-                reason=u'%s: Automember rule not found' % hostgroup1.cn)):
+    @pytest.mark.parametrize(
+        "automember_grp, grp", [
+            ("automember_group", "group1"),
+            ("automember_hostgroup", "hostgroup1")
+        ])
+    def test_delete_with_nonexistent_group(self, automember_grp, grp, request):
+        """ Try to delete a rule with non-existent group/ hostgroup """
+        grp = request.getfixturevalue(grp)
+        automember_grp = request.getfixturevalue(automember_grp)
+        grp.ensure_missing()
```

Comment:
It should be automember_grp.ensure_missing().
The function can be simplified and use only automember_grp + request parameters (grp is not useful here, grp.cn can be replaced with automember_grp.cn).